develop/2017-06-20-rodolfoviana-expenses-on-car-rental.py

```diff
@@ -0,0 +1,254 @@
+# 
+# The first scenario (2 x `std value`) shows we have 99 outliers. The second one (3 x `std value`) gives us 22 outliers.
+
+# In[11]:
+
+full_report_car.net_values.describe()
+
+
+# 
+
+# In[12]:
+
+from scipy.stats import normaltest
+
+def normaltest_pvalue(values):
+    if len(values) >= 20:
+        return normaltest(values).pvalue
+    else:
+        return 1
+
+net_values_by_cnpj_1 = full_report_car.groupby('cnpj')['net_values']     .agg([len, np.mean, np.std, normaltest_pvalue])     .sort_values('len', ascending=False)     .reset_index()
+net_values_by_cnpj_1['threshold'] = net_values_by_cnpj_1['mean'] +     2 * net_values_by_cnpj_1['std']
+applicants_per_cnpj = full_report_car.groupby('cnpj')['applicant_id']     .aggregate(lambda x: len(set(x))).reset_index()     .rename(columns={'applicant_id': 'congresspeople'})
+net_values_by_cnpj_1 = pd.merge(net_values_by_cnpj_1, applicants_per_cnpj)
+net_values_by_cnpj_1.head()
+
+
+# In[13]:
+
+len(net_values_by_cnpj_1.query('normaltest_pvalue < .05')) / len(net_values_by_cnpj_1)
+
+
+# In[14]:
+
+def normaltest_pvalue(values):
+    if len(values) >= 20:
+        return normaltest(values).pvalue
+    else:
+        return 1
+
+net_values_by_cnpj_2 = full_report_car.groupby('cnpj')['net_values']     .agg([len, np.mean, np.std, normaltest_pvalue])     .sort_values('len', ascending=False)     .reset_index()
+net_values_by_cnpj_2['threshold'] = net_values_by_cnpj_2['mean'] +     3 * net_values_by_cnpj_2['std']
+applicants_per_cnpj = full_report_car.groupby('cnpj')['applicant_id']     .aggregate(lambda x: len(set(x))).reset_index()     .rename(columns={'applicant_id': 'congresspeople'})
+net_values_by_cnpj_2 = pd.merge(net_values_by_cnpj_2, applicants_per_cnpj)
+net_values_by_cnpj_2.head()
+
+
+# In[15]:
+
+len(net_values_by_cnpj_2.query('normaltest_pvalue < .05')) / len(net_values_by_cnpj_2)
+
+
+# In[16]:
+
+data_with_threshold_1 = pd.merge(full_report_car, net_values_by_cnpj_1, on='cnpj')     .sort_values('net_values', ascending=False)
+
+
+# In[17]:
+
+data_with_threshold_2 = pd.merge(full_report_car, net_values_by_cnpj_2, on='cnpj')     .sort_values('net_values', ascending=False)
+
+
+# In[18]:
+
+outliers_1 = data_with_threshold_1.query('(congresspeople >= 1) & (len >= 20) & (net_values > threshold)')
+print(len(outliers_1), outliers_1['net_values'].sum())
+
+
+# In[19]:
+
+outliers_2 = data_with_threshold_2.query('(congresspeople >= 1) & (len >= 20) & (net_values > threshold)')
+print(len(outliers_2), outliers_2['net_values'].sum())
+
+
+# #### Getting toll expenses
+
+# In[20]:
+
+companies_toll = companies[companies['main_activity_code'] == '52.21-4-00']
+
+
+# In[21]:
+
+data_toll = data[data['subquota_description'] == 'Taxi, toll and parking']

```

Comment:
same thing here, this object could have a more meaningful name like: `reimbursements_toll`